### PR TITLE
Added an exception for quest locations

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: 

* **Server Mode**: Both

* **Description of Pull Request**: 

Added an exception for quest locations for memorial.
Updated the docs.

### CODE REVIEW

File: src/map/quest.cpp [hunk 1]
```diff
@@ -726,6 +726,12 @@ void quest_update_objective(struct map_session_data *sd, struct mob_data* md)
 					objective_check++;
 				if (qi->objectives[j]->mapid < 0 || (qi->objectives[j]->mapid == sd->bl.m && md->spawn != nullptr))
 					objective_check++;
+				else if (qi->objectives[j]->mapid >= 0) {
+					struct map_data *mapdata = map_getmapdata(sd->bl.m);
+
+					if (mapdata->instance_id && map_getmapdata(mapdata->instance_src_map)->m == qi->objectives[j]->mapid)
```
Comment (Lemongrass3110) [inline src/map/quest.cpp:732]:
```suggestion
					if (mapdata->instance_id && mapdata->instance_src_map == qi->objectives[j]->mapid)
```
This should already be the mapid.

File: db/import-tmpl/quest_db.yml [hunk 2]
```diff
@@ -42,7 +42,7 @@
 #       Element     Monster element target (default All). Valid elements are Dark, Earth, Fire, Ghost, Holy, Neutral, Poison, Undead, Water, Wind, All.
 #       MinLevel    Minimum monster level target. If not supplied but MaxLevel defined, MinLevel is 1. Set to 0 to ignore MinLevel on import. (Default: 0)
 #       MaxLevel    Maximum monster level target. Set to 0 to ignore MaxLevel on import. (Default: 0)
-#       Location    Name of hunting location from mapindex. (Default any location)
+#       Location    Name of hunting location from mapindex. Except in memorial, only permanent monsters increase the counter. (Default any location)
```
Comment (Lemongrass3110) [inline db/import-tmpl/quest_db.yml:45]:
```suggestion
#       Location    Mapname where kills are counted. Outside of instances only permanent monsters increase the counter. (Default any location)
```

File: src/map/quest.cpp (hunk 1 above)
Comment (Atemo) [reply src/map/quest.cpp]: True!

**Review by Lemongrass3110 — APPROVED** [review]

File: db/import-tmpl/quest_db.yml [hunk 3]
```diff
@@ -42,7 +42,7 @@
 #       Element     Monster element target (default All). Valid elements are Dark, Earth, Fire, Ghost, Holy, Neutral, Poison, Undead, Water, Wind, All.
 #       MinLevel    Minimum monster level target. If not supplied but MaxLevel defined, MinLevel is 1. Set to 0 to ignore MinLevel on import. (Default: 0)
 #       MaxLevel    Maximum monster level target. Set to 0 to ignore MaxLevel on import. (Default: 0)
-#       Location    Name of hunting location from mapindex. (Default any location)
+#       Location    Mapname where kills are counted. Outside of instances only permanent monsters increase the counter. (Default any location)
```
Comment (aleos89) [inline db/import-tmpl/quest_db.yml:45]:
```suggestion
#       Location    Map name where kills are counted. (Default any location)
#                   On normal maps only kills of permanently spawned monsters increase the counter.
#                   On instance maps all monster kills increase the counter.
```

File: db/pre-re/quest_db.yml [hunk 4]
```diff
@@ -42,7 +42,7 @@
 #       Element     Monster element target (default All). Valid elements are Dark, Earth, Fire, Ghost, Holy, Neutral, Poison, Undead, Water, Wind, All.
 #       MinLevel    Minimum monster level target. If not supplied but MaxLevel defined, MinLevel is 1. Set to 0 to ignore MinLevel on import. (Default: 0)
 #       MaxLevel    Maximum monster level target. Set to 0 to ignore MaxLevel on import. (Default: 0)
-#       Location    Name of hunting location from mapindex. (Default any location)
+#       Location    Mapname where kills are counted. Outside of instances only permanent monsters increase the counter. (Default any location)
```
Comment (aleos89) [inline db/pre-re/quest_db.yml:45]:
```suggestion
#       Location    Map name where kills are counted. (Default any location)
#                   On normal maps only kills of permanently spawned monsters increase the counter.
#                   On instance maps all monster kills increase the counter.
```

File: db/quest_db.yml [hunk 5]
```diff
@@ -42,7 +42,7 @@
 #       Element     Monster element target (default All). Valid elements are Dark, Earth, Fire, Ghost, Holy, Neutral, Poison, Undead, Water, Wind, All.
 #       MinLevel    Minimum monster level target. If not supplied but MaxLevel defined, MinLevel is 1. Set to 0 to ignore MinLevel on import. (Default: 0)
 #       MaxLevel    Maximum monster level target. Set to 0 to ignore MaxLevel on import. (Default: 0)
-#       Location    Name of hunting location from mapindex. (Default any location)
+#       Location    Mapname where kills are counted. Outside of instances only permanent monsters increase the counter. (Default any location)
```
Comment (aleos89) [inline db/quest_db.yml:45]:
```suggestion
#       Location    Map name where kills are counted. (Default any location)
#                   On normal maps only kills of permanently spawned monsters increase the counter.
#                   On instance maps all monster kills increase the counter.
```

File: db/re/quest_db.yml [hunk 6]
```diff
@@ -42,7 +42,7 @@
 #       Element     Monster element target (default All). Valid elements are Dark, Earth, Fire, Ghost, Holy, Neutral, Poison, Undead, Water, Wind, All.
 #       MinLevel    Minimum monster level target. If not supplied but MaxLevel defined, MinLevel is 1. Set to 0 to ignore MinLevel on import. (Default: 0)
 #       MaxLevel    Maximum monster level target. Set to 0 to ignore MaxLevel on import. (Default: 0)
-#       Location    Name of hunting location from mapindex. (Default any location)
+#       Location    Mapname where kills are counted. Outside of instances only permanent monsters increase the counter. (Default any location)
```
Comment (aleos89) [inline db/re/quest_db.yml:45]:
```suggestion
#       Location    Map name where kills are counted. (Default any location)
#                   On normal maps only kills of permanently spawned monsters increase the counter.
#                   On instance maps all monster kills increase the counter.
```

File: doc/yaml/db/quest_db.yml [hunk 7]
```diff
@@ -25,7 +25,7 @@
 #       Element     Monster element target (default All). Valid elements are Dark, Earth, Fire, Ghost, Holy, Neutral, Poison, Undead, Water, Wind, All.
 #       MinLevel    Minimum monster level target. If not supplied but MaxLevel defined, MinLevel is 1. Set to 0 to ignore MinLevel on import. (Default: 0)
 #       MaxLevel    Maximum monster level target. Set to 0 to ignore MaxLevel on import. (Default: 0)
-#       Location    Name of hunting location from mapindex. (Default any location)
+#       Location    Mapname where kills are counted. Outside of instances only permanent monsters increase the counter. (Default any location)
```
Comment (aleos89) [inline doc/yaml/db/quest_db.yml:28]:
```suggestion
#       Location    Map name where kills are counted. (Default any location)
#                   On normal maps only kills of permanently spawned monsters increase the counter.
#                   On instance maps all monster kills increase the counter.
```

**Review by aleos89 — APPROVED** [review]
Just expanding on the documentation a bit.

**Pull request merged**